docs/csharp/tutorials/console-webapiclient.md

```diff
@@ -1,388 +1,354 @@
 
-You're almost done with this section. Now that you've converted the JSON to C# objects, let's display
-the name of each repository. Replace the lines that read:
+   The output is a long display of JSON text.
 
-```csharp
-var msg = await stringTask;   //**Deleted this
-Console.Write(msg);
-```
+## Deserialize the JSON Result
 
-with the following:
+The following steps convert the JSON response into C# objects. You use the <xref:System.Text.Json.JsonSerializer?displayProperty=nameWithType> class to deserialize JSON into objects.
 
-```csharp
-foreach (var repo in repositories)
-    Console.WriteLine(repo.name);
-```
+1. Create a file named *repo.cs* and put the following code in it. This code defines a class to represent the JSON object returned from the GitHub API:
 
-Compile and run the application. It will print the names of the repositories that are part of the
-.NET Foundation.
+   ```csharp
+   using System;
+   
+   namespace WebAPIClient
+   {
+       public class Repository
+       {
+           public string name { get; set; }
+       }
+   }
+   ```
 
-## Controlling Serialization
+   You'll use this class to display a list of repository names.
 
-Before you add more features, let's address the `name` property by using the `[JsonPropertyName]` attribute. Make
-the following changes to the declaration of the `name` field in repo.cs:
+   The JSON for a repository object contains dozens of properties,but only the `name` property will be deserialized. The serializer automatically ignores JSON properties for which there is no match in the target class. This feature makes it easier to create types that work with only a subset of fields in a large JSON packet.
 
-```csharp
-[JsonPropertyName("name")]
-public string Name { get; set; }
-```
+   The C# convention is to capitalize the first letter of property names, but the `name` property here starts with a lowercase letter because that matches exactly what's in the JSON. Later you'll see how to use C# property names that don't match the JSON property names.
 
-To use `[JsonPropertyName]` attribute, you will need to add the <xref:System.Text.Json.Serialization> namespace to the `using` directives:
+1. Use the serializer to convert JSON into C# objects. Replace the call to
+<xref:System.Net.Http.HttpClient.GetStringAsync(System.String)> in the `ProcessRepositories` method with the following lines:
 
-```csharp
-using System.Text.Json.Serialization;
-```
+   ```csharp
+   var streamTask = client.GetStreamAsync("https://api.github.com/orgs/dotnet/repos");
+   var repositories = await JsonSerializer.DeserializeAsync<List<Repository>>(await streamTask);
+   ```
 
-This change means you need to change the code that writes the name of each repository in program.cs:
+   The new code calls <xref:System.Net.Http.HttpClient.GetStreamAsync(System.String)> instead of <xref:System.Net.Http.HttpClient.GetStringAsync(System.String)>. This serializer method uses a stream instead of a string as its source.
```

Comment:
Perfectly written. I'd write it as:

```suggestion
   The updated code replaces <xref:System.Net.Http.HttpClient.GetStringAsync(System.String)> with <xref:System.Net.Http.HttpClient.GetStreamAsync(System.String)>. This serializer method uses a stream instead of a string as its source.
```